x-pack/plugins/security_solution/server/lib/detection_engine/migrations/delete_migration.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ElasticsearchClient, SavedObjectsClientContract } from 'src/core/server';
+import { isMigrationDeleted, isMigrationFailed, isMigrationPending } from './helpers';
+import { applyMigrationCleanupPolicy } from './migration_cleanup';
+import { SignalsMigrationSO } from './saved_objects_schema';
+import { updateMigrationSavedObject } from './update_migration_saved_object';
+
+/**
+ * "Deletes" a completed migration:
+ *   * soft-deletes the migration SO
```

Comment:
What's the advantage of leaving the migration SO in a soft-deleted state here? In my mind we'd want the task document and the migration SO to have the same lifetime, but it seems that the SOs would live on indefinitely.

Reply:
updated to do a hard-delete in https://github.com/elastic/kibana/pull/85690/commits/a1d2d6a67ee8bb31107156b7d15016842b945010